mem0/vector_stores/neptune_analytics.py

```diff
@@ -0,0 +1,497 @@
+
+    def __init__(
+        self,
+        endpoint: str,
+        collection_name: str,
+        embedding_model_dims: int = 1536,
+    ):
+        """
+        Initialize the Neptune Analytics vector store.
+
+        Args:
+            endpoint (str): Neptune Analytics endpoint in format 'neptune-graph://<graphid>'.
+            collection_name (str): Name of the collection to store vectors.
+            embedding_model_dims (int, optional): Dimension of embedding vectors. Defaults to 1536.
+            
+        Raises:
+            ValueError: If endpoint format is invalid.
+            ImportError: If langchain_aws is not installed.
+        """
+
+        if not endpoint.startswith("neptune-graph://"):
+            raise ValueError("Please provide 'endpoint' with the format as 'neptune-graph://<graphid>'.")
+
+        graph_id = endpoint.replace("neptune-graph://", "")
+        self.graph = NeptuneAnalyticsGraph(graph_id)
+        self.collection_name = self._COLLECTION_PREFIX + collection_name
+        self.embedding_model_dims = embedding_model_dims
+
+    
+    def create_col(self, name, vector_size, distance):
+        """
+        Create a collection (no-op for Neptune Analytics).
+        
+        Neptune Analytics supports dynamic collections that are created implicitly
+        when vectors are inserted, so this method performs no operation.
+        
+        Args:
+            name: Collection name (unused).
+            vector_size: Vector dimension (unused).
+            distance: Distance metric (unused).
+        """
+        pass
+
+    
+    def insert(self, vectors: List[list],
+        payloads: Optional[List[Dict]] = None,
+        ids: Optional[List[str]] = None):
+        """
+        Insert vectors into the collection.
+        
+        Creates or updates nodes in Neptune Analytics with vector embeddings and metadata.
+        Uses MERGE operation to handle both creation and updates.
+        
+        Args:
+            vectors (List[list]): List of embedding vectors to insert.
+            payloads (Optional[List[Dict]]): Optional metadata for each vector.
+            ids (Optional[List[str]]): Optional IDs for vectors. Generated if not provided.
+        """
+
+        para_list = []
+        for index, data_vector in enumerate(vectors):
+            if payloads:
+                payload = payloads[index]
+                payload[self._FIELD_LABEL] = self.collection_name
+            else:
+                payload = {}
+            para_list.append(dict(
+                node_id=ids[index] if ids else str(uuid.uuid4()),
+                properties=payload,
+                embedding=data_vector,
+            ))
+
+        para_map_to_insert = {"rows": para_list}
+
+        query_string = (f"""
+            UNWIND $rows AS row
+            MERGE (n
+            :{self.collection_name}
+             {{`~id`: row.node_id}})
+            ON CREATE SET n = row.properties, n.updated_at = toString(timestamp())
```

Comment:
in other vector stores, we create the timestamp locally and send it (so that its in the pacific/us timezone).  We should be consistent?  What do you think?